src/app/profiles/profiles.editor.modal.ts

```diff
@@ -50,131 +48,101 @@ export class ProfilesEditorModalComponent {
     selector: any;
     public tinymce_plugin: TinyMCEPlugin;
     constructor(
+        public profileService: ProfileService,
         public rmm: RMM,
         private location: Location,
         public snackBar: MatSnackBar,
         public dialog_ref: MatDialogRef<ProfilesEditorModalComponent>,
         public draftDeskservice: DraftDeskService,
-        @Inject(MAT_DIALOG_DATA) public data: any
+        @Inject(MAT_DIALOG_DATA) public identity: Identity
     ) {
         this.tinymce_plugin = new TinyMCEPlugin();
-        if (data && data.type) {
-            this.type = data.type;
-            delete data.type;
-        }
-        if (data.profile && data.profile.email) {
-            this.set_localpart(data);
+        if (identity.email) {
+            this.set_localpart(identity);
         }
-        if (!data || !Object.keys(data).length || !data.profile) {
-            data = { profile: {} };
+        if (!identity || !Object.keys(identity).length) {
+            identity = new Identity;
             const self = this;
-            data.profile.name = ['first_name', 'last_name'].map((attr) => {
-                return self.rmm.me.data[attr];
+            identity.name = ['first_name', 'last_name'].map((attr) => {
+                return self.profileService.me[attr];
             }).join(' ');
         }
-        this.data = data;
-        if (this.data.profile.is_signature_html) {
+        this.identity = identity;
+        if (this.identity.is_signature_html) {
             this.init_tinymce();
         } else {
-            this.data.profile.is_signature_html = false;
+            this.identity.is_signature_html = false;
         }
-        this.check_reply_to(this.data);
+        this.check_reply_to(this.identity);
     }
-    check_reply_to(data) {
-        if (data && data.profile.email && data.profile.reply_to &&
-            data.profile.reply_to !== data.profile.email) {
+    check_reply_to(identity) {
+        if (identity && identity.email && identity.reply_to &&
+            identity.reply_to !== identity.email) {
             this.is_different_reply_to = true;
```

Comment:
Preferably this would be a utility function, like `isDifferentReplyTo(identity)`, but I understand you might not want to change it since its how the whole component works.